gapis/trace/android/trace.go

```diff
@@ -242,26 +234,49 @@ func (t *androidTracer) GetTraceTargetNode(ctx context.Context, uri string, icon
 		pkg,
 		"",
 	}
+
+	var firstActivity *pkginfo.Activity
+	var defaultAction string
 	for _, a := range instPkg.Activities {
 		if len(a.Actions) > 0 {
 			r.Children = append(r.Children, fmt.Sprintf("%s/%s", pkg, a.Name))
-			for _, act := range a.Actions {
-				if act.IsLaunch {
-					defaultAction = fmt.Sprintf("%s:%s/%s", act.Name, pkg, a.Name)
-				}
+			if act := findBestAction(a.Actions); act != nil && act.IsLaunch {
+				defaultAction = fmt.Sprintf("%s:%s/%s", act.Name, pkg, a.Name)
+			}
+			if firstActivity == nil {
+				firstActivity = a
 			}
 		}
 	}
 
-	if len(instPkg.Activities) == 1 {
-		if len(instPkg.Activities[0].Actions) == 1 {
-			defaultAction = fmt.Sprintf("%s:%s/%s", instPkg.Activities[0].Actions[0].Name, pkg, instPkg.Activities[0].Name)
+	if defaultAction == "" && len(r.Children) == 1 {
```

Comment:
Yes, we only want to do this if no launch action was found and the application has only one child (activity). I'm looking at the child count, because we filter activities without any actions in the loop above.